src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.hpp

```diff
@@ -0,0 +1,188 @@
+    size_t get_vec_length() const;
+
+    mutable std::vector<uint32_t> aux_vec_idxs;
+    mutable std::vector<uint32_t> aux_gpr_idxs;
+
+    dnnl::impl::cpu::aarch64::jit_generator* h;
+    dnnl::impl::cpu::aarch64::cpu_isa_t host_isa_;
+    ov::element::Type exec_prc_;
+    const float alpha;
+
+    emitter_in_out_map in_out_type_;
+
+    virtual void prepare_table();
+    virtual void register_table_entries() {}
+
+    void load_table_addr() const { h->adr(p_table, *l_table.get()); }
+
+    // we accept only 32bit hexadecimal table values to avoid any rounding
+    using table_entry_val_t = uint32_t;
+    using table_entry_offset_t = size_t; // offsets are in bytes wrt p_table
+    using table_entry_bcast_t = bool; // true => bcast value
+
+    struct table_entry_t {
+        table_entry_val_t val;
+        table_entry_bcast_t bcast;
+    };
+    struct mapped_table_entry_t {
+        table_entry_offset_t off;
+        table_entry_val_t val;
+        table_entry_bcast_t bcast;
+    };
+
+    mutable Xbyak_aarch64::XReg p_table;
+    mutable std::shared_ptr<Xbyak_aarch64::Label> l_table;
+
+    virtual void emit_impl(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs) const = 0;
+
+    virtual void emitter_preamble(const std::vector<size_t>& in_idxs,
+                                  const std::vector<size_t>& out_idxs,
+                                  const std::vector<size_t>& pool_aux_vec_idxs,
+                                  const std::vector<size_t>& pool_aux_gpr_idxs) const;
+
+    virtual void emitter_postamble() const;
+
+    void store_context(const std::unordered_set<size_t>& ignore_registers) const;
+
+    void restore_context(const std::unordered_set<size_t>& ignore_registers) const;
+
+    using table_t = std::multimap<std::string, table_entry_t>;
+    using mapped_table_t = std::multimap<std::string, mapped_table_entry_t>;
+
+    mapped_table_t entry_map_;
+
+    Xbyak_aarch64::AdrImm table_val(std::string key, size_t key_off_val_shift = 0) const {
+        const int32_t off = table_off(key, key_off_val_shift);
+        return Xbyak_aarch64::ptr(p_table, off);
+    }
+
+    Xbyak_aarch64::AdrNoOfs table_val2(std::string key, size_t key_off_val_shift = 0) const {
+        const int32_t off = table_off(key, key_off_val_shift);
+        h->add_imm(h->X_DEFAULT_ADDR, p_table, off, h->X_TMP_0);
+        return Xbyak_aarch64::ptr(h->X_DEFAULT_ADDR);
+    }
+
+    void push_arg_entry_of(const std::string key, const table_entry_val_t val, const bool broadcast) {
+        mapped_table_entry_t te {0, val, broadcast};
+        entry_map_.insert(std::make_pair(key, te));
+    }
+
+    void push_entries_of(const table_t &t) {
+        for (auto it = t.begin(); it != t.end(); it++) {
+            auto key = (*it).first;
+            auto te = (*it).second; // copy values from table
+            push_arg_entry_of(key, te.val, te.bcast);
+        }
+    }
+
+    static inline int32_t get_asimd_vector_length() {
```

Comment:
We already have `get_vec_length` method. No need to create another one

Reply:
done, thanks